plugin/xcodebuild.vim

```diff
@@ -0,0 +1,48 @@
+command! XcodebuildBuild call <sid>build()
+command! XcodebuildTest call <sid>test()
+
+let s:plugin_path = expand("<sfile>:p:h:h")
+
+function! s:build()
+  let cmd = s:base_command()  . s:xcpretty()
+  call s:run_command(cmd)
+endfunction
+
+function! s:test()
+  let cmd =  s:base_command() . ' -sdk iphonesimulator test' . s:xcpretty()
+  call s:run_command(cmd)
+endfunction
+
+function! s:run_command(cmd)
+  execute '!' . a:cmd
+endfunction
+
+function! s:base_command()
+  return 'xcodebuild ' . s:build_target() . ' ' . s:scheme()
+endfunction
+
+function! s:build_target()
+  let xcworkspaceFile = globpath(expand('.'), '*.xcworkspace')
+  if empty(xcworkspaceFile)
+    return '-project ' . s:project_file()
+  else
+    return '-workspace ' . xcworkspaceFile
+  endif
+endfunction
+
+function! s:project_file()
+  return globpath(expand('.'), '*.xcodeproj')
+endfunction
+
+function! s:scheme()
+  let scheme = "silent ! 'sh " . s:plugin_path . "/bin/find_scheme.sh \"" . s:project_file() . "\""
```

Comment:
You could also replace the `.` with `source`, which I think is not POSIX-compliant but is more readable and works everywhere you care about like sh/bash/zsh.